src/sentry/integrations/slack/integration.py

```diff
@@ -67,18 +68,77 @@
     aspects={"alerts": [setup_alert]},
 )
 
+_default_logger = logging.getLogger(__name__)
+
 
 class SlackIntegration(SlackNotifyBasicMixin, IntegrationInstallation):
+    _FLAGS_KEY: str = "TOGGLEABLE_FLAGS"
+    _ISSUE_ALERTS_THREAD_FLAG: str = "ISSUE_ALERTS_THREAD_FLAG"
+    _METRIC_ALERTS_THREAD_FLAG: str = "METRIC_ALERTS_THREAD_FLAG"
+    _SUPPORTED_FLAGS_WITH_DEFAULTS: dict[str, bool] = {
+        _ISSUE_ALERTS_THREAD_FLAG: True,
+        _METRIC_ALERTS_THREAD_FLAG: True,
+    }
+
     def get_client(self) -> SlackClient:
         return SlackClient(integration_id=self.model.id)
 
-    def get_config_data(self) -> Mapping[str, str]:
+    def get_config_data(self) -> Mapping[str, Any]:
+        base_data = super().get_config_data()
+
+        # Add installationType key to config data
         metadata_ = self.model.metadata
         # Classic bots had a user_access_token in the metadata.
         default_installation = (
             "classic_bot" if "user_access_token" in metadata_ else "workspace_app"
         )
-        return {"installationType": metadata_.get("installation_type", default_installation)}
+        base_data["installationType"] = metadata_.get("installation_type", default_installation)
+
+        # Add missing toggleable feature flags
```

Comment:
is this so we don't have to do a migration every time we add a flag?

Reply:
yup! Realistically we should pull the features into it's own table and just link it to the integration installation. This way we you add a new flag the default is already there